xapian-core/include/xapian/weight.h

```diff
@@ -409,6 +409,11 @@ class XAPIAN_VISIBILITY_DEFAULT TfIdfWeight : public Weight {
     /// The factor to multiply with the weight.
     double factor;
 
+    /// Parameters slope and delta in the Piv+ normalization weighting formula.
+    /// Default values: 0.2 and 1.0 respectively.
+    double param_slope = 0.2;
+    double param_delta = 1.0;
+
     TfIdfWeight * clone() const;
```

Comment:
@ojwb I've made the changes you suggested and all tests pass under `make check` and gdb but valgrind reports some errors -- mainly two errors spread over a no. of statements. 
First one is again: 
```
Conditional jump or move depends on uninitialised value(s)
at 0x4F937C5: MultiMatch::get_mset(unsigned int, unsigned int, unsigned int, Xapian::MSet&, Xapian::Weight::Internal&, Xapian::MatchDecider const*, Xapian::KeyMaker const*) (multimatch.cc:802)
==1046==    by 0x4E92250: Xapian::Enquire::Internal::get_mset(unsigned int, unsigned int, unsigned int, Xapian::RSet const*, Xapian::MatchDecider const*) const (omenquire.cc:548)
==1046==    by 0x4E92573: Xapian::Enquire::get_mset(unsigned int, unsigned int, unsigned int, Xapian::RSet const*, Xapian::MatchDecider const*) const (omenquire.cc:906)
==1046==    by 0x5B9118: test_tfidfweight4() (api_weight.cc:682)
```
and similar one by `api_weight.cc:691`. Also, another error which I think occurs due to something's wrong with `TfIdfWeight::clone()`:
```
Uninitialised value was created by a heap allocation
==1046==    at 0x4C2E0EF: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==1046==    by 0x4FC7F32: Xapian::TfIdfWeight::clone() const (tfidfweight.cc:82)
==1046==    by 0x4F8D789: LocalSubMatch::open_post_list(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int, double, bool, QueryOptimiser*, bool) (localsubmatch.cc:300)
==1046==    by 0x4F8C7D9: LocalSubMatch::get_postlist(MultiMatch*, unsigned int*) (localsubmatch.cc:204)
==1046==    by 0x4F92AF0: MultiMatch::get_mset(unsigned int, unsigned int, unsigned int, Xapian::MSet&, Xapian::Weight::Internal&, Xapian::MatchDecider const*, Xapian::KeyMaker const*) (multimatch.cc:419)
==1046==    by 0x4E92250: Xapian::Enquire::Internal::get_mset(unsigned int, unsigned int, unsigned int, Xapian::RSet const*, Xapian::MatchDecider const*) const (omenquire.cc:548)
==1046==    by 0x4E92573: Xapian::Enquire::get_mset(unsigned int, unsigned int, unsigned int, Xapian::RSet const*, Xapian::MatchDecider const*) const (omenquire.cc:906)
==1046==    by 0x5B9975: test_tfidfweight4() (api_weight.cc:691)
```
Could you please tell me what I might be missing here?